### PR TITLE
Graph: add methods to get vertices and edges in a range of IDs

### PR DESCRIPTION
**Pull request by joeferner**
Accumulo: add methods to get the tablet splits

@srfarley

### CODE REVIEW

File: core/src/main/java/org/vertexium/GraphBase.java [hunk 1]
```diff
@@ -114,6 +114,28 @@ protected boolean isIncluded(Vertex v) {
...
+    }
+
+    @Override
+    public Iterable<Vertex> getVerticesInRange(Range idRange, EnumSet<FetchHint> fetchHints, Authorizations authorizations) {
+        return getVerticesInRange(idRange, fetchHints, null, authorizations);
+    }
+
+    @Override
+    public Iterable<Vertex> getVerticesInRange(final Range idRange, EnumSet<FetchHint> fetchHints, Long endTime, Authorizations authorizations) {
+        LOGGER.warn("Performing scan of all vertices! Override getVerticesInRange.");
```
Comment (srfarley) [inline core/src/main/java/org/vertexium/GraphBase.java]: I find these warning messages really annoying when running the unit tests. Can we just log each message once?

File: core/src/test/java/org/vertexium/RangeTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+        range = new Range(null, "c");
+        assertTrue(range.isInRange("b"));
+        assertFalse(range.isInRange("c"));
+
+        range = new Range("b", null);
+        assertFalse(range.isInRange("az"));
+        assertTrue(range.isInRange("b"));
+    }
+}
```
Comment (srfarley) [inline core/src/test/java/org/vertexium/RangeTest.java]: There should be tests for `Range#equals()` and `Range#hashCode()`.

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java [hunk 3]
```diff
@@ -1546,6 +1591,42 @@ public Connector getConnector() {
         return connector;
     }
 
+    public Iterable<Range> listVerticesTableSplits() {
```
Comment (srfarley) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java]: Can these methods be package-protected?

Comment (joeferner) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java]: No, I need them in my spark code.

File: core/src/main/java/org/vertexium/GraphBase.java (hunk 1 above)
Comment (joeferner) [inline core/src/main/java/org/vertexium/GraphBase.java]: I agree. I'll make a new PR for that.

File: core/src/test/java/org/vertexium/RangeTest.java (hunk 2 above)
Comment (joeferner) [inline core/src/test/java/org/vertexium/RangeTest.java]: done

File: core/src/test/java/org/vertexium/RangeTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        assertFalse(range.isInRange("c"));
+
+        range = new Range("b", null);
+        assertFalse(range.isInRange("az"));
+        assertTrue(range.isInRange("b"));
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @Test
+    public void testHashCode() {
```
Comment (srfarley) [inline core/src/test/java/org/vertexium/RangeTest.java]: Yeah, maybe `hashCode` doesn't need tests.

Comment (joeferner) [inline core/src/test/java/org/vertexium/RangeTest.java]: It does test the handling of nullness but that's about it.

**Pull request merged**